examples/stream.wasm/emscripten.cpp

```diff
@@ -103,11 +103,11 @@ void stream_main(size_t index) {
 
             {
                 const int n_segments = whisper_full_n_segments(ctx);
-                for (int i = n_segments - 1; i < n_segments; ++i) {
-                    const char * text = whisper_full_get_segment_text(ctx, i);
+                if (n_segments > 0) {
```

Comment:
This will only process the last segment. We want to process all segments

```suggestion
                for (int i = n_segments - 1; i < n_segments && n_segments > 0; ++i) {
```

Reply:
`single_segment` is set to true so it will not output more than one segment. Furthermore, the original code's `for` loop will only iterate once. Because it starts from `n-1`, it will only process the last segment.

Reply:
Oh, I missed that 👍